drivers/saul/init_devs/auto_init_hmc5883l.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/*
+ * @ingroup     sys_auto_init_saul
+ * @brief       Auto initialization of HMC5883L 3-axis digital compass
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ */
+
+#ifdef MODULE_HMC5883L
+
+#include "assert.h"
+#include "log.h"
+#include "saul_reg.h"
+
+#include "hmc5883l.h"
+#include "hmc5883l_params.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define HMC5883L_NUM    (sizeof(hmc5883l_params) / sizeof(hmc5883l_params[0]))
```

Comment:
```suggestion
#define HMC5883L_NUM    ARRAY_SIZE(hmc5883l_params)
```

Reply:
Yeah, the PR is older than the `ARRAY_SIZE` macro.